guides/v2.1/howdoi/checkout/checkout-add-custom-carrier.md

```diff
@@ -0,0 +1,298 @@
+---
+layout: tutorial
+group: how-do-i
+subgroup:
+title: Add custom shipping carrier
+subtitle: Customize Checkout
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+menu_order: 5
+level3_subgroup: checkout-tutorial
+functional_areas:
+  - Checkout
+---
+
+This topic describes how to add a custom shipping carrier.
+
+To add a new shipping carrier to the Magento checkout:
+
+1. [Create a new module](#create-module)
+2. [Add the carrier configuration](#create-configuration)
+3. [Create the carrier model](#carrier-model)
+4. [Enable the module](#enable-module)
+
+## Step 1: Create a new module {#create-module}
```

Comment:
I'm curious if the sections here are enough for someone to create a new module. There are no instructions, just source code snippets.

Reply:
Should there be a "To create a new module" list with simple steps on how to utilize each of these source code snippets?

Reply:
> I'm curious if the sections here are enough for someone to create a new module. There are no instructions, just source code snippets.

@shrielenee , this guide is for developers. Working source code is the best instruction. Please believe me. 😊

Reply:
> Should there be a "To create a new module" list with simple steps on how to utilize each of these source code snippets?

No, Let's leave all steps as they are.